pkg/policy/api/rule_validation.go

```diff
@@ -381,6 +381,41 @@ func (pr *PortRule) sanitize(ingress bool) error {
 		}
 	}
 
+	listener := pr.Listener
+	if listener != nil {
+		if listener.Name == "" {
+			return fmt.Errorf("Listener must have name")
+		}
+		if listener.EnvoyConfig == nil {
+			return fmt.Errorf("Listener must have envoyConfig (%s)",
+				listener.Name)
+		}
+		if listener.EnvoyConfig.Name == "" {
+			return fmt.Errorf("Listener envoyConfig must have name (%s)", listener.Name)
+		}
```

Comment:
[1] will allow you to drop this line.

Reply:
Should I drop all other (manual) validation in favor of letting k8s worry about it as well?

Reply:
Removed also other validation that is already done vie kubebuilder validation